src/dryad.lisp

```diff
@@ -50,6 +50,13 @@
     :documentation "A map ADDRESS -> BOOLEAN which records whether a `BLOSSOM-NODE' has begun participating in matches."))
   (:documentation "PROCESS responsible for the injection and ejection of nodes from the blossom algorithm."))
 
+(defgeneric create-discovery-message (dryad &key channels-to-try id)
+  (:documentation "Generic `MESSAGE-DISCOVERY' constructor that dispatches on the type of `DRYAD' that is creating it. `CHANNELS-TO-TRY' is the same as the `DISCOVERY-MESSAGE' slot, and `ID' is the same as the `DISCOVER-MESSAGE' slot."))
```

Comment:
Consider renaming the defstruct constructor by passing `:constructor %make-message-discovery`, naming this `make-discovery-message`, and having it call `%make-message-discovery`.

Also consider not inverting the word order used elsewhere.